### DISCUSSION TITLE
Generic OpenID Connect / OAuth2 auth provider not requesting a scope `groups`

### DISCUSSION THREAD

**Discussion by vojbarzz** [requarks/wiki#7232] [Error / Bug Report]
Hello, having configured Authelia using Generic OpenID auth provider and using option `Map Groups` and having a `Groups Claim` configured to `groups`. Unfortunately the scope does not contains `goups` :-(
```scope=openid%20openid%20profile%20email```
the question is why the openid is there twice, if it is not some implementation issue. If I modify authorization request to contain also a group scope gtoup mapping works as expected.

Does anybody know how to make it working?

![image](https://github.com/requarks/wiki/assets/16055634/0a5a6fb3-b136-48e8-b5ac-89ffa309e2c5)

**Comment by robotron2084** [comment]
I've similarly come across this issue, but I'm using Authentik. This feels very much like a bug in WikiJs. Map Groups is basically broken because it is never requesting the scope, so it never sets the group. I also noticed that `openid` is in the request twice, which leads me to believe that there is a copy/paste error within the system.

Update: After spending a bit of time on this, I'm able to get this to work if i use `groups` as my scope instead of my own scope which i just called `wikijs`. I pulled down the source and looked at  `server/modules/authentication/oidc/authentication.js` and saw that it had some hardcoded checks against that scope.  I tried making the `wikijs` scope as a string, as an array, an array as a string, none of that worked. As soon as I changed the scope to `groups` then it finally mapped to the given role. Which I don't think this is how its supposed to work. Its supposed to look at the `groupsClaim` group, look at the roles that are defined in the Wiki, then look at the `groups` group and give access based on the delta. But it doesn't seem to pull down any custom scopes so it dies, but using a 'builtin' scope like `groups` does work.

**Reply by Hallo5000** [reply]
I just finished my setup with authentik ("Generic OpenID Connect / OAuth2") too and for anyone having issues:
- the groups array is actually part of authentiks profile field (so no need for custom mappings) 
- "Assign to group" can be left empty (otherwise everyone will have a "default" group) 
- any group assigned by the authentication provider must exist in WikiJs beforehand (so unfortunately you have to create every group you want to use manually in WikiJs) 
- also you can skip WikiJs Login Screen under Security > Login > Bypass Login Screen

**Comment by eddeT** [comment]
This issue persists as per my testing from yesterday.

My setup: https://github.com/authelia/authelia/discussions/11325

**Comment by M1N0RM1N3R** [comment]
I'm in the same boat with Pocket ID. What I did was recreate it as a "Generic OAuth2" strategy instead. With that, I set the scope to `profile email groups`, then manually tell it what fields to use from the userinfo endpoint. Here's my final config: (Basically everything but the client secret is public knowledge if you know where to find it, and if this were to be private forever, I'd have all this behind Tailscale anyway.)
<img width="1260" height="523" alt="image" src="https://github.com/user-attachments/assets/680f22f2-b6ce-4d97-9c62-25aae36e4121" />
<img width="1256" height="914" alt="image" src="https://github.com/user-attachments/assets/622ccc0f-a04b-4103-9ee3-e16d9aa6daa9" />
<img width="1262" height="712" alt="image" src="https://github.com/user-attachments/assets/b89bc8ce-333c-42bc-9899-fe678bed123e" />
Just don't forget to update the callback URL with the new strategy ID. :P
